contracts/src/tokens/memecoin.cairo

```diff
@@ -297,7 +301,22 @@ mod UnruggableMemecoin {
             )
         }
 
-        /// Constructor logic.
+        /// Internal function to prevent the multicall buys
+        ///
+        /// This check make sure that an address won't be
+        /// able to make the multicall buys, as we are keeping
+        /// track of the transaction hash.
+        ///
+        /// # Arguments
+        /// * `sender` - The contract address of the caller/sender.
+        #[inline(always)]
+        fn _check_and_update_tx_hash(ref self: ContractState, sender: ContractAddress) {
+            let tx_hash: felt252 = get_tx_info().unbox().transaction_hash;
+            assert(self.tx_hash_tracker.read(sender) != tx_hash, 'Multi calls not allowed');
+            self.tx_hash_tracker.write(sender, tx_hash);
```

Comment:
@abdelhamidbakhta aren't we increasing transfer costs by two by adding another storage var? Can we think of another mechanism?